deluge/ui/gtkui/mainwindow.py

```diff
@@ -109,6 +109,24 @@ def first_show(self):
             while gtk.events_pending():
                 gtk.main_iteration(False)
             self.vpaned.set_position(self.initial_vpaned_position)
+        elif deluge.common.windows_check():
+            # Realize window to create a handle for wndproc.
+            self.window.realize()
+
+        if deluge.common.windows_check():
+            # Handle win32 WM_QUERYENDSESSION message so GUI can be shutdown cleanly.
```

Comment:
Lets try and drop pywin32 module usage, so I'm thinking:

```python
from ctypes.wintypes import HWND, WPARAM, LPARAM, UINT, WINFUNCTYPE
from ctypes import windll, c_wchar_p
CallWindowProc = windll.user32.CallWindowProcW
SetWindowLong = windll.user32.SetWindowLongW

LRESULT = LPARAM
WNDPROC = WINFUNCTYPE(LRESULT, HWND, UINT, WPARAM, LPARAM)

WM_QUERYENDSESSION = 17
GWL_WNDPROC = -4
```

References:

- https://github.com/KellyZ/USB-Clipboard/blob/master/clipboard.py
- https://github.com/rene-aguirre/pywinusb/blob/master/pywinusb/hid/wnd_hook_mixin.py